math/mean/mean_test.go

```diff
@@ -0,0 +1,47 @@
+package mean
+
+import (
+	"testing"
+)
+
+func TestMean(t *testing.T){
+	testCases := []struct{
+		name string
+		testValues []float64
+		average float64
+	}{
+		{
+			name: "All 0s", 
+			testValues: []float64{0,0,0,0,0},
+			average: 0,
+		},
+		{
+			name: "With integer values",
+			testValues: []float64{1,2,3,4,5},
+			average: 3.0,
+		},
+		{
+			name: "With negative values",
+			testValues: []float64{-1,2,-3,4,5},
+			average: 1.4,
+		},
+		{
+			name: "With floating values",
+			testValues: []float64{1.1,2.2,3.3,4.4,5.5},
+			average: 3.3,
+		},
```

Comment:
A another test case with no values inside.

```
testValues: []float64{},
```

to make sure you cover this edge case.